### PR TITLE
safeweb: detect mux handler conflicts

### PR DESCRIPTION
**Pull request by awly**
When both muxes match, and one of them is a wildcard "/" pattern (which is common in browser muxes), choose the more specific pattern. If both are non-wildcard matches, there is a pattern overlap, so return an error.

Updates https://github.com/tailscale/corp/issues/8027

### CODE REVIEW

**Review by patrickod — APPROVED** [review]
Lush. Thank you!

File: safeweb/http.go [hunk 1]
```diff
@@ -208,10 +173,70 @@ func NewServer(config Config) (*Server, error) {
...
+	_, ap := s.APIMux.Handler(r)
+	switch {
+	case bp == "" && ap != "": // APIMux match
+		s.serveAPI(w, r)
+	case bp != "" && ap == "": // BrowserMux match
+		s.serveBrowser(w, r)
+	case bp == "" && ap == "": // neither match
+		http.NotFound(w, r)
+	case bp != "" && ap != "":
+		// Both muxes match the path. This can be because:
```
Comment (creachadair) [inline safeweb/http.go:201]: Perhaps rather than special-casing "/", we should prefer the longer match?

Comment (awly) [reply safeweb/http.go:201]: We should, but I'm worried it's not as simple as a longer string match. See https://go.dev/blog/routing-enhancements#precedence
OK if I leave it as TODO for now?

Comment (creachadair) [reply safeweb/http.go:201]: For sure, better even as is. I believe it should be safe as the matcher returns path bounded strings, but we should definitely make sure that's true.

**Pull request merged**